openassessment/xblock/static/js/src/lms/oa_response.js

```diff
@@ -468,14 +470,22 @@ export class ResponseView {
         const currentResponse = view.response();
         const currentResponseEqualsSaved = currentResponse.every((element, index) => element === savedResponse[index]);
         if (currentResponseEqualsSaved) {
-          view.saveEnabled(false);
-          const msg = gettext('This response has been saved but not submitted.');
-          view.saveStatus(msg);
+          const msg = gettext('Draft saved!');
+          view.saveStatus(msg, this.ICON_SAVED);
           view.baseView.srReadTexts([msg]);
+
+          // Disable error
+          this.baseView.toggleActionError('save', null);
+          view.errorOnLastSave = false;
         }
       }).fail((errMsg) => {
-        view.saveStatus(gettext('Error'));
-        view.baseView.toggleActionError('save', errMsg);
+        // Debounce error banner, this won't capture new errors, but will keep
+        // us from defocusing text area, allowing user to continue to edit their
+        // response.
+        if (!view.errorOnLastSave) {
```

Comment:
The comment explains, but running `toggleActionError` defocuses the text field. This causes annoying page refreshes *and* makes it hard to keep typing because the text box loses focus every few seconds.

Reply:
^^ It will either get stuck in the "Error" or "Draft Saving" state, depending on when the async events fire, but this felt like a small enough issue that I didn't want to sink a bunch of time into fixing it.